### PR TITLE
PR to develop branch for Microplanning v0.1 (changes in plan-service, resource-generator service)

### PR DESCRIPTION
**Pull request by Priyanka-eGov**

### CODE REVIEW

**Comment by Priyanka-eGov** [comment]
@CodeRabbit please generate a summary and list the key features in this PR.

**Comment by sathishp-eGov** [comment]
@coderabbitai review

**Review by kavi-egov — CHANGES_REQUESTED** [review]

File: health-services/census-service/src/main/java/digit/repository/rowmapper/StatusCountRowMapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class StatusCountRowMapper implements ResultSetExtractor<Map<String, Integer>> {
+
+    @Override
+    public Map<String, Integer> extractData(ResultSet rs) throws SQLException, DataAccessException {
+        Map<String, Integer> statusCountMap = new HashMap<>();
```
Comment (kavi-egov) [inline health-services/census-service/src/main/java/digit/repository/rowmapper/StatusCountRowMapper.java:18]: There is a method in jdbc itself that will give you the map, we don't need this

File: health-services/census-service/src/main/java/digit/service/CensusService.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    public CensusResponse create(CensusRequest request) {
+
+        // Validate census create request
+        validator.validateCreate(request);
+
+        // Enrich census create request
+        enrichment.enrichCreate(request);
+
+        // Enrich timeframe for previous census
+        timeframeEnrichment.enrichPreviousTimeframe(request);
```
Comment (kavi-egov) [inline health-services/census-service/src/main/java/digit/service/CensusService.java:56]: What is time Frame enrichment, need comments

**Review by kavi-egov — APPROVED** [review]

**Pull request merged**